### DISCUSSION TITLE
The alternative for `useQuereies` `keepPreviousData` in v5

### DISCUSSION THREAD

**Discussion by RichardPanP** [TanStack/query#6521] [Q&A] [answered]
Hi, I'm upgrading `@tanstack/react-query` to v5 and noticed that `keepPreviousData` has been deprecated. And there's a note that `useQueries` would not receive `previousData` in the `placeholderData`. Is there any way if I still hope to keep the previous data in `useQueries`?

**Comment by TkDodo** [comment, marked as answer]
keepPreviousData on useQueries wasn't really working in v4 with dynamic length arrays. It's too difficult to find out which observer was the "previous one" when lengths change. we removed it for that reason.

**Reply by dylangrandmont** [reply]
Are there any patterns or code examples you can recommend to replace this behaviour for those of us performing the migration to v5 who relied on `keepPreviousData` inside `useQueries`? For instance [this blog](https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose) was quite useful to migrating away from the callbacks which were removed from `useQuery`; is there any equivalent write up for this topic?

**Reply by TkDodo** [reply]
if the queries are known - use multiple `useQuery` calls. If they are truly dynamic, try to split them up into separate components.
